### PR TITLE
Fix for mapping rule with same type and different name

### LINKED ISSUE #285: ros1_bridge does not bridge service if service field name changes, even if mapped in mapping_rules.yaml
## Bug report
I have found that if a service field name changes between ROS1 and ROS2, even if the change is mapped in a `mapping_rules.yaml`, does not bridge the service. 

**Required Info:**

- Operating System:
  - Ubuntu 20.04.
- Installation type:
  - Source
- Version or commit hash:
  - Tag 0.9.4
- DDS implementation:
  - Default (Fast RTPS)
- Client library (if applicable):
  - N/A

#### Steps to reproduce issue
I have created a repository containing the example code here: https://github.com/RaAndOn/bridge_failure, but I will also include the files here:

```
.
├── ros1_ws
│   └── src
│       └── example_interfaces
│           ├── CMakeLists.txt
│           ├── package.xml
│           └── srv
│               ├── CustomServiceFails.srv
│               └── CustomServiceSucceeds.srv
└── ros2_ws
    └── src
        └── example_interfaces
            ├── CMakeLists.txt
            ├── mapping_rules.yaml
            ├── package.xml
            └── srv
                ├── CustomServiceFails.srv
                └── CustomServiceSucceeds.srv
```
ROS1 CMakeLists.txt

```
cmake_minimum_required(VERSION 2.8.3)

project(example_interfaces)

set(MSG_DEPS std_msgs)

set(BUILD_DEPS ${MSG_DEPS} message_generation)

set(RUNTIME_DEPS ${MSG_DEPS} message_runtime)

find_package(catkin REQUIRED COMPONENTS ${BUILD_DEPS})

add_service_files(DIRECTORY srv FILES
    CustomServiceFails.srv
    CustomServiceSucceeds.srv
)

generate_messages(DEPENDENCIES ${MSG_DEPS})

catkin_package(CATKIN_DEPENDS ${RUNTIME_DEPS})

```
ROS1 package.xml
```
<?xml version="1.0" ?>
<package format="2">
  <name>example_interfaces</name>
  <version>0.0.1</version>
  <description>
    This pacakge contains a single service to demonstrate a bug in the ros1_bridge
  </description>
  <maintainer email="joshua.raanan@gmail.com">Joshua Ra'anan</maintainer>

  <license>BSD</license>
  <buildtool_depend>catkin</buildtool_depend>

  <depend>std_msgs</depend>

  <build_depend>message_generation</build_depend>
  <exec_depend>message_runtime</exec_depend>

  <export>
  </export>
</package>
```
ROS1 CustomServiceFails.srv
```
float64 same_field_name
---
float64 differentFieldName
```
ROS1 CustomServiceSucceeds.srv
```
float64 fine_name
---
float64 also_fine_name
```
ROS2 CMakeLists.txt
```
cmake_minimum_required(VERSION 3.5)
project(example_interfaces)

if(NOT CMAKE_CXX_STANDARD)
  set(CMAKE_CXX_STANDARD 14)
endif()
if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
  add_compile_options(-Wall -Wextra -Wpedantic)
endif()

set(MSG_DEPS
  std_msgs
)

set(BUILD_DEPS
  ament_cmake
  rosidl_default_generators
  ${MSG_DEPS}
)

foreach(DEP ${BUILD_DEPS})
  find_package(${DEP} REQUIRED)
endforeach()

find_package(rosidl_default_generators REQUIRED)

rosidl_generate_interfaces(${PROJECT_NAME}
    srv/CustomServiceFails.srv
    srv/CustomServiceSucceeds.srv
DEPENDENCIES ${MSG_DEPS}
)

install(
  FILES mapping_rules.yaml
  DESTINATION share/${PROJECT_NAME})

ament_export_dependencies(rosidl_default_runtime)

ament_package()
```
ROS2 package.xml
```
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
<package format="3">
  <name>example_interfaces</name>
  <version>0.0.1</version>
  <description>
    This pacakge contains a single service to demonstrate a bug in the ros1_bridge
  </description>
  <maintainer email="joshua.raanan@gmail.com">Joshua Ra'anan</maintainer>

  <license>BSD</license>
  
  <buildtool_depend>ament_cmake</buildtool_depend>

  <depend>std_msgs</depend>

  <buildtool_depend>rosidl_default_generators</buildtool_depend>
  <exec_depend>rosidl_default_runtime</exec_depend>

  <member_of_group>rosidl_interface_packages</member_of_group>

  <export>
    <build_type>ament_cmake</build_type>
    <ros1_bridge mapping_rules="mapping_rules.yaml"/>
  </export>

</package>
```
ROS2 mapping_rules.yaml
```
-
    ros1_package_name: "example_interfaces"
    ros1_service_name: "CustomServiceFails"
    ros2_package_name: "example_interfaces"
    ros2_service_name: "CustomServiceFails"
    request_fields_1_to_2:
        same_field_name: "same_field_name"
    response_fields_1_to_2:
        differentFieldName: "different_field_name"
```
ROS2 CustomServiceFails.srv
```
float64 same_field_name
---
float64 different_field_name
```
ROS2 CustomServiceSucceeds.srv
```
float64 fine_name
---
float64 also_fine_name
```

The build instructions are the default instructions:
```
cd . <ros1-workspace-path>/
. /opt/ros/noetic/setup.bash
catkin build
cd . <ros2-workspace-path>/
. /opt/ros/noetic/setup.bash
colcon build
. <ros1-workspace-path>/example_interfaces/devel/setup.bash
. <ros2-workspace-path>/example_interfaces/install/local_setup.bash
colcon build --packages-select ros1_bridge --cmake-force-configure
```
Checking the `--print-pairs` shows that the `CustomServiceFails.srv` is not mapped.
#### Expected behavior
The expected behavior is that both `example_interfaces/CustomServiceSucceeds` and `example_interfaces/CustomServiceFails` are bridged.

#### Actual behavior
What actually happens is that only `example_interfaces/CustomServiceSucceeds` is bridged and `example_interfaces/CustomServiceFails` is not bridged.
```
Supported ROS 2 <=> ROS 1 service type conversion pairs:
  - 'diagnostic_msgs/srv/AddDiagnostics' (ROS 2) <=> 'diagnostic_msgs/AddDiagnostics' (ROS 1)
  - 'diagnostic_msgs/srv/SelfTest' (ROS 2) <=> 'diagnostic_msgs/SelfTest' (ROS 1)
  - 'example_interfaces/srv/CustomServiceSucceeds' (ROS 2) <=> 'example_interfaces/CustomServiceSucceeds' (ROS 1)
  - 'gazebo_msgs/srv/BodyRequest' (ROS 2) <=> 'gazebo_msgs/BodyRequest' (ROS 1)
```

### PR DESCRIPTION
**Pull request by iuhilnehc-ynos**
Fixes #285

Signed-off-by: Chen Lihui <lihui.chen@sony.com>

### CODE REVIEW

**Comment by tgaspar** [comment]
I have implemented your changes on my end and it does work as intended.

Removing the `or ros1_name != ros2_name`) check seems to be a reasonable thing to do there. 

Thanks for taking the time to do this.

**Comment by iuhilnehc-ynos** [comment]
@sloretz, Could you review it when you become free?

File: ros1_bridge/__init__.py [hunk 1]
```diff
@@ -606,7 +606,7 @@ def determine_common_services(
                 ros2_type = str(ros2_fields[direction][i].type)
                 ros1_name = ros1_field[1]
                 ros2_name = ros2_fields[direction][i].name
-                if ros1_type != ros2_type or ros1_name != ros2_name:
+                if ros1_type != ros2_type:
```
Comment (fujitatomoya) [inline ros1_bridge/__init__.py:609]: according to https://github.com/ros2/ros1_bridge/blob/master/doc/index.rst, this is not specification. it should not be checked here.

**Review by fujitatomoya — APPROVED** [review]
lgtm

File: ros1_bridge/__init__.py [hunk 2]
```diff
@@ -606,7 +606,7 @@ def determine_common_services(
                 ros2_type = str(ros2_fields[direction][i].type)
                 ros1_name = ros1_field[1]
                 ros2_name = ros2_fields[direction][i].name
-                if ros1_type != ros2_type or ros1_name != ros2_name:
```
Comment (RaAndOn) [inline ros1_bridge/__init__.py:609]: I don't think you can just get rid of the field name checking. Rather if the field names don't match, shouldn't the code compare the issue against the mapping yaml?

Comment (iuhilnehc-ynos) [reply ros1_bridge/__init__.py:609]: Thank you for your review, I'll check it.

Comment (iuhilnehc-ynos) [reply ros1_bridge/__init__.py:609]: @RaAndOn

Could you help review https://github.com/ros2/ros1_bridge/pull/293/commits/873c2057ce820bd3f2c36245a54daad9ddfa860c again?

**Comment by iuhilnehc-ynos** [comment]
There is a friendly person @ipa-hsd to resolve the issue https://github.com/ros2/ros1_bridge/issues/285, close this PR.
I'll keep an eye on that and test it.

**Pull request closed without merging**